src/main/java/com/gmail/goosius/siegewar/metadata/ResidentMetaDataController.java

```diff
@@ -32,14 +35,87 @@ public static int getNationRefundAmount(Resident resident) {
 		return 0;
 	}
 
-	public static void setNationRefundAmount(Resident resident, int nationRefundAmount) {
+	public static void clearNationRefund(Resident resident) {
 		IntegerDataField idf = (IntegerDataField) refundAmount.clone();
 		if (resident.hasMeta(idf.getKey())) {
 			resident.removeMetaData(idf);
-			if (nationRefundAmount != 0)
+		}
+	}
+
+	public static void addNationRefundAmount(Resident resident, int nationRefundAmount) {
+		IntegerDataField idf = (IntegerDataField) refundAmount.clone();
+		if (nationRefundAmount > 0) {
+			if (resident.hasMeta(idf.getKey())) {
+				int updatedRefundAmount = getNationRefundAmount(resident) + nationRefundAmount;
+				resident.removeMetaData(idf);
+				resident.addMetaData(new IntegerDataField("siegewar_nationrefund", updatedRefundAmount, "Nation Refund"));

```

Comment:
Throughout this class you're using removeMetaData and then addMetaData, instead of setMetaData. Was this to get around it not saving the resident when setMetaData was used?

Reply:
- No, it wasn't for that
- I just did't try using set
- I'll change it to that now

Reply:
remember to resident.save() when using setMetaData()

Reply:
- I had a look, but the only metadata set method Resident has is setMetadata(String), inherited from Towny object
- It doesn't take a CustomDataField like addMetaData(CustomDataField)
- Should I leave as it is?, or something else?

Reply:
leave it be, the way you've done it here isn't wrong either way, I'll look into the setMetaData.